### PR TITLE
MULE-14039: MetadataService starts Sources when resolving Metadata

### PR DESCRIPTION
**Pull request by juandesi**

### CODE REVIEW

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/metadata/SourceMetadataTestCase.java [hunk 1]
```diff
@@ -64,4 +66,11 @@ public void getSourceDynamicOutputMetadata() throws Exception {
     assertExpectedOutput(componentMetadata.getModel(), personType, typeLoader.load(StringAttributes.class));
     assertThat(componentMetadata.getMetadataAttributes().getKey().get(), is(PERSON_METADATA_KEY));
   }
+
+  @Test
+  public void sourcesMustNotStartWhenResolvingMetadata() {
+    final MetadataResult<ComponentMetadataDescriptor<SourceModel>> result = getComponentDynamicMetadata(PERSON_METADATA_KEY);
+    assertThat(result.isSuccess(), is(true));
+    assertThat(MetadataSource.STARTED, is(false));
```
Comment (estebanwasinger) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/metadata/SourceMetadataTestCase.java]: this doesn't work, classloading issues

Comment (alegmarra) [reply modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/metadata/SourceMetadataTestCase.java]: so.. he didn't run the test?

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/metadata/SourceMetadataTestCase.java [hunk 2]
```diff
@@ -64,4 +66,11 @@ public void getSourceDynamicOutputMetadata() throws Exception {
     assertExpectedOutput(componentMetadata.getModel(), personType, typeLoader.load(StringAttributes.class));
     assertThat(componentMetadata.getMetadataAttributes().getKey().get(), is(PERSON_METADATA_KEY));
   }
+
+  @Test
+  public void sourcesMustNotStartWhenResolvingMetadata() {
```
Comment (alegmarra) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/metadata/SourceMetadataTestCase.java]: after asserting it is stopped, also test for starting it

Comment (juandesi) [reply modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/metadata/SourceMetadataTestCase.java]: why?  I don't see the point

**Pull request merged**